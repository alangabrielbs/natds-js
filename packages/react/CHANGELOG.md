# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.5.4](https://github.com/natura-cosmeticos/natds-js/compare/v1.5.3...v1.5.4) (2021-03-16)

**Note:** Version bump only for package @naturacosmeticos/natds-react
