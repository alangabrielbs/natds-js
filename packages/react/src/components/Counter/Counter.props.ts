import { Size } from '@naturacosmeticos/natds-themes'

type CounterSizes = keyof Pick<Size, 'semi' | 'medium' >

export interface CounterProps {
  
  disabled?: boolean

  size?: CounterSizes

  label?: string

  readOnly?: boolean

  maxValue?: number

  minValue?: number

  onIncrement: () => void

  onDecrement: () => void,
 
  value: number

  onChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>

}
