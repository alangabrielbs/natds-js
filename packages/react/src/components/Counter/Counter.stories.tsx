import React from 'react'
import { Story, Meta } from '@storybook/react'
import { Counter, CounterProps } from '.'

const componentStatus = `
---

**NOTE FOR UXs**: This component is available in the following variants:

  - ❌ \`standard\`

With the following attribute status:

- ❌ **Label**
- ❌ **Size**
- ❌ **Disabled**

---
`

export default {
  title: 'Components/Counter',
  component: Counter,
  parameters: {
    componentSubtitle: 'Reduce input effort for fields with values that deviate little from the default by allowing users to increase or decrease the number in a single button press.',
    docs: { description: { component: componentStatus } },
    actions: { argTypesRegex: '^on.*' }
  }
} as Meta


export const Playground: Story<CounterProps> = (args) => { 
  const [value, setValue] = React.useState(args.value)

  const handleIncrement = () => setValue(value + 1)
  const handleDecrement = () => setValue(value - 1)

  return (
    <div>
      <Counter {...args} onIncrement={handleIncrement} onDecrement={handleDecrement} value={value} />
    </div>
  )
}

Playground.args = {value : 0, label: "label" }