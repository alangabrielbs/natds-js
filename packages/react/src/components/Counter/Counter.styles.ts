import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { CounterProps } from './Counter.props'

const styles = createUseStyles((theme: Theme) => ({

    buttonBase: {
      backgroundColor: "transparent",
      outline: 0,
      appearance: "none",
      height: ({size}: CounterProps) => size && theme.size[size],
      width:  ({size}: CounterProps) => size && size === "semi" ? theme.size[size] : theme.size.semiX,
      borderRadius: 4,
    },

    inputBase: {
      textAlign: 'center',
      height: ({size}: CounterProps) => size && theme.size[size],
      width:  ({size}: CounterProps) => size && size === "semi" ? theme.size[size] : theme.size.semiX
    }
    
}))

export default styles
