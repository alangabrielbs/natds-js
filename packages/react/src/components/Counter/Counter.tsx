/* eslint-disable max-lines-per-function */
import React from 'react'
import { CounterProps } from './Counter.props'
import Label from '../_subcomponents/Label'
import styles from './Counter.styles'
import Button from '../Button'

const Counter = ({
  disabled,
  label,
  maxValue = 99,
  minValue = 0,
  onChange,
  onDecrement,
  onIncrement,
  size = 'semiX',
  testID,
  value = 0,
}: CounterProps) => {
  const { buttonGroup, inputBase } = styles({ size, disabled })

  const maxReached = value >= maxValue
  const minReached = value <= minValue

  return (
    <div data-testid={testID}>
      {label && <Label color="mediumEmphasis" label={label} />}
      <div className={buttonGroup}>
        <Button
          variant="text"
          text="-"
          onClick={onDecrement}
          disabled={disabled || minReached}
          size={size}
        />
        <input
          value={value}
          onChange={onChange}
          className={inputBase}
          readOnly
        />
        <Button
          variant="text"
          text="+"
          onClick={onIncrement}
          disabled={disabled || maxReached}
          size={size}
        />
      </div>
    </div>
  )
}

export default Counter
