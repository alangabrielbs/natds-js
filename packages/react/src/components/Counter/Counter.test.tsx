import React from 'react'
import { fireEvent } from '@testing-library/react'
import { CounterProps } from './Counter.props'
import Counter from './Counter'
import renderWithTheme from '../../helpers/renderWithTheme'

const defaultProps: CounterProps = {
  onChange: () => '',
  onIncrement: () => '',
  onDecrement: () => '',
  value: 0
}

describe('Counter', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<Counter {...defaultProps} />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly with label', () => {
    const { styles, component } = renderWithTheme(<Counter {...defaultProps} label="label" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should call onIncrement when is not disabled', () => {
    const onIncrementMock = jest.fn()
    const { component } = renderWithTheme(
      <Counter {...defaultProps} onIncrement={onIncrementMock} />
    )
    fireEvent.click(component.getByTestId('onIncrement-btn'))

    expect(onIncrementMock).toHaveBeenCalledTimes(1)
  })

  it('should call onDecrement when is not disabled', () => {
    const onDecrementMock = jest.fn()
    const { component } = renderWithTheme(
      <Counter {...defaultProps} onDecrement={onDecrementMock} value={1} />
    )
    fireEvent.click(component.getByTestId('onDecrement-btn'))

    expect(onDecrementMock).toHaveBeenCalledTimes(1)
  })
})
