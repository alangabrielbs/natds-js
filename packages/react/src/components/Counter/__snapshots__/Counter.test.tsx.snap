// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render correctly with default props 1`] = `
Array [
  ".buttonGroup-0-2-1 {
  width: fit-content;
  border: 1px solid #bbbbbb;
  margin-top: 4px;
  border-radius: 4px;
}
.buttonGroup-0-2-1 button {}
.inputBase-0-2-2 {
  color: #000000;
  width: 40px;
  cursor: default;
  outline: none;
  border-top: none;
  text-align: center;
  border-left: 1px solid #bbbbbb;
  border-right: 1px solid #bbbbbb;
  border-bottom: none;
  background-color: transparent;
}
.buttonGroup-d0-0-2-3 {}
.buttonGroup-d0-0-2-3 button {
  width: 32px;
}
.inputBase-d1-0-2-4 {
  height: 40px;
}
.sharedRippleEffect-0-2-11 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-12 {
  outline: none;
  position: relative;
  align-self: start;
}
.wrapper-0-2-12:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-12:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.ripple-0-2-13 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-14 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-15 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-16 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-17 {
    cursor: default;
    display: inline-block;
  }
  .wrapper-d0-0-2-17:focus:after {  }
  .wrapper-d0-0-2-17:hover:after {
    background-color: #000000;
    opacity: 0;
  }
  .ripple-d1-0-2-18 {
    background-color: #000000;
    height: 0;
    left: 0;
    top: 0;
    transition: scale opacity 300ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-19 {
    overflow: hidden;
  }
  .rippleActive-d3-0-2-20 {
    animation: keyframes-ripple-0-2-16 300ms;
  }
  .wrapper-d4-0-2-24 {
    cursor: pointer;
    display: inline-block;
  }
  .wrapper-d4-0-2-24:focus:after {
    background-color: #000000;
  }
  .wrapper-d4-0-2-24:hover:after {
    background-color: #000000;
    opacity: 0;
  }
  .ripple-d5-0-2-25 {
    background-color: #000000;
    height: 0;
    left: 0;
    top: 0;
    transition: scale opacity 300ms;
    width: 0;
  }
  .rippleContainer-d6-0-2-26 {
    overflow: hidden;
  }
  .rippleActive-d7-0-2-27 {
    animation: keyframes-ripple-0-2-16 300ms;
  }
  .button-0-2-5 {
    cursor: pointer;
    outline: 0;
    overflow: hidden;
    position: relative;
    border-radius: 4px;
  }
  .button-0-2-5:after {
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    content: \\" \\";
    opacity: 0;
    position: absolute;
    background-color: #000000;
  }
  .button-0-2-5:hover:not([disabled]):after {
    opacity: 0.04;
  }
  .button-0-2-5:focus:not([disabled]):after {
    opacity: 0.12;
  }
  .button-0-2-5:disabled {
    cursor: default;
    box-shadow: none;
  }
  .labelContainer-0-2-6 {
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .label-0-2-7 {
    overflow: hidden;
    font-size: 14px;
    text-align: center;
    font-family: Roboto, sans-serif;
    font-weight: 500;
    user-select: none;
    white-space: nowrap;
    text-overflow: ellipsis;
    letter-spacing: 1.23px;
    text-transform: uppercase;
  }
  .button-d0-0-2-8 {
    background-color: transparent;
    border: 0;
    box-shadow: none;
    height: 40px;
    padding-left: 8px;
    padding-right: 8px;
    width: auto;
  }
  .button-d0-0-2-8:disabled {
    background-color: transparent;
  }
  .labelContainer-d1-0-2-9 {
    flex-direction: row-reverse;
  }
  .label-d2-0-2-10 {
    color: #777777;
  }
  .button-d3-0-2-21 {
    background-color: transparent;
    border: 0;
    box-shadow: none;
    height: 40px;
    padding-left: 8px;
    padding-right: 8px;
    width: auto;
  }
  .button-d3-0-2-21:disabled {
    background-color: transparent;
  }
  .labelContainer-d4-0-2-22 {
    flex-direction: row-reverse;
  }
  .label-d5-0-2-23 {
    color: #333333;
  }",
  <div>
    <div>
      <div
        class="buttonGroup-0-2-1 buttonGroup-d0-0-2-3"
      >
        <div
          class="wrapper-0-2-12 wrapper-d0-0-2-17"
          data-testid="ripple-wrapper"
          tabindex="-1"
        >
          <div
            class="rippleContainer-0-2-14 rippleContainer-d2-0-2-19"
          >
            <div
              class="ripple-0-2-13 ripple-d1-0-2-18 "
              data-testid="ripple-animation"
            />
          </div>
          <button
            class="button-0-2-5 button-d0-0-2-8"
            data-testid="onDecrement-btn"
            disabled=""
            type="button"
          >
            <span
              class="labelContainer-0-2-6 labelContainer-d1-0-2-9"
            >
              <span
                class="label-0-2-7 label-d2-0-2-10"
              >
                -
              </span>
            </span>
          </button>
        </div>
        <input
          class="inputBase-0-2-2 inputBase-d1-0-2-4"
          readonly=""
          value="0"
        />
        <div
          class="wrapper-0-2-12 wrapper-d4-0-2-24"
          data-testid="ripple-wrapper"
          tabindex="-1"
        >
          <div
            class="rippleContainer-0-2-14 rippleContainer-d6-0-2-26"
          >
            <div
              class="ripple-0-2-13 ripple-d5-0-2-25 "
              data-testid="ripple-animation"
            />
          </div>
          <button
            class="button-0-2-5 button-d3-0-2-21"
            data-testid="onIncrement-btn"
            type="button"
          >
            <span
              class="labelContainer-0-2-6 labelContainer-d4-0-2-22"
            >
              <span
                class="label-0-2-7 label-d5-0-2-23"
              >
                +
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Counter should render correctly with label 1`] = `
Array [
  ".label-0-2-5 {
  font-family: Roboto, sans-serif;
  letter-spacing: 0.22px;
}
.label-d0-0-2-6 {
  font-size: 14px;
  color: #777777;
}
.buttonGroup-0-2-1 {
  width: fit-content;
  border: 1px solid #bbbbbb;
  margin-top: 4px;
  border-radius: 4px;
}
.buttonGroup-0-2-1 button {}
.inputBase-0-2-2 {
  color: #000000;
  width: 40px;
  cursor: default;
  outline: none;
  border-top: none;
  text-align: center;
  border-left: 1px solid #bbbbbb;
  border-right: 1px solid #bbbbbb;
  border-bottom: none;
  background-color: transparent;
}
.buttonGroup-d0-0-2-3 {}
.buttonGroup-d0-0-2-3 button {
  width: 32px;
}
.inputBase-d1-0-2-4 {
  height: 40px;
}
.sharedRippleEffect-0-2-13 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-14 {
  outline: none;
  position: relative;
  align-self: start;
}
.wrapper-0-2-14:focus:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  opacity: 0.16;
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.wrapper-0-2-14:hover:after {
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: \\" \\";
  z-index: -1;
  position: absolute;
  border-radius: 50%;
}
.ripple-0-2-15 {
  opacity: 0;
  position: absolute;
  transform: translate(-50%, -50%);
  border-radius: 50%;
}
.rippleContainer-0-2-16 {
  inset: 0;
  position: absolute;
}
.rippleActive-0-2-17 {
  z-index: 99;
}
@keyframes keyframes-ripple-0-2-18 {
  0% {
    transform: translate(-50%, -50%) scale(0);
  }
  75% {
    opacity: 0.12;
    transform: translate(-50%, -50%) scale(1);
  }
  100% {
    opacity: 0;
    z-index: 0;
  }
}
  .wrapper-d0-0-2-19 {
    cursor: default;
    display: inline-block;
  }
  .wrapper-d0-0-2-19:focus:after {  }
  .wrapper-d0-0-2-19:hover:after {
    background-color: #000000;
    opacity: 0;
  }
  .ripple-d1-0-2-20 {
    background-color: #000000;
    height: 0;
    left: 0;
    top: 0;
    transition: scale opacity 300ms;
    width: 0;
  }
  .rippleContainer-d2-0-2-21 {
    overflow: hidden;
  }
  .rippleActive-d3-0-2-22 {
    animation: keyframes-ripple-0-2-18 300ms;
  }
  .wrapper-d4-0-2-26 {
    cursor: pointer;
    display: inline-block;
  }
  .wrapper-d4-0-2-26:focus:after {
    background-color: #000000;
  }
  .wrapper-d4-0-2-26:hover:after {
    background-color: #000000;
    opacity: 0;
  }
  .ripple-d5-0-2-27 {
    background-color: #000000;
    height: 0;
    left: 0;
    top: 0;
    transition: scale opacity 300ms;
    width: 0;
  }
  .rippleContainer-d6-0-2-28 {
    overflow: hidden;
  }
  .rippleActive-d7-0-2-29 {
    animation: keyframes-ripple-0-2-18 300ms;
  }
  .button-0-2-7 {
    cursor: pointer;
    outline: 0;
    overflow: hidden;
    position: relative;
    border-radius: 4px;
  }
  .button-0-2-7:after {
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    content: \\" \\";
    opacity: 0;
    position: absolute;
    background-color: #000000;
  }
  .button-0-2-7:hover:not([disabled]):after {
    opacity: 0.04;
  }
  .button-0-2-7:focus:not([disabled]):after {
    opacity: 0.12;
  }
  .button-0-2-7:disabled {
    cursor: default;
    box-shadow: none;
  }
  .labelContainer-0-2-8 {
    display: flex;
    align-items: center;
    justify-content: center;
  }
  .label-0-2-9 {
    overflow: hidden;
    font-size: 14px;
    text-align: center;
    font-family: Roboto, sans-serif;
    font-weight: 500;
    user-select: none;
    white-space: nowrap;
    text-overflow: ellipsis;
    letter-spacing: 1.23px;
    text-transform: uppercase;
  }
  .button-d0-0-2-10 {
    background-color: transparent;
    border: 0;
    box-shadow: none;
    height: 40px;
    padding-left: 8px;
    padding-right: 8px;
    width: auto;
  }
  .button-d0-0-2-10:disabled {
    background-color: transparent;
  }
  .labelContainer-d1-0-2-11 {
    flex-direction: row-reverse;
  }
  .label-d2-0-2-12 {
    color: #777777;
  }
  .button-d3-0-2-23 {
    background-color: transparent;
    border: 0;
    box-shadow: none;
    height: 40px;
    padding-left: 8px;
    padding-right: 8px;
    width: auto;
  }
  .button-d3-0-2-23:disabled {
    background-color: transparent;
  }
  .labelContainer-d4-0-2-24 {
    flex-direction: row-reverse;
  }
  .label-d5-0-2-25 {
    color: #333333;
  }",
  <div>
    <div>
      <label
        class="label-0-2-5 label-d0-0-2-6"
      >
        label
      </label>
      <div
        class="buttonGroup-0-2-1 buttonGroup-d0-0-2-3"
      >
        <div
          class="wrapper-0-2-14 wrapper-d0-0-2-19"
          data-testid="ripple-wrapper"
          tabindex="-1"
        >
          <div
            class="rippleContainer-0-2-16 rippleContainer-d2-0-2-21"
          >
            <div
              class="ripple-0-2-15 ripple-d1-0-2-20 "
              data-testid="ripple-animation"
            />
          </div>
          <button
            class="button-0-2-7 button-d0-0-2-10"
            data-testid="onDecrement-btn"
            disabled=""
            type="button"
          >
            <span
              class="labelContainer-0-2-8 labelContainer-d1-0-2-11"
            >
              <span
                class="label-0-2-9 label-d2-0-2-12"
              >
                -
              </span>
            </span>
          </button>
        </div>
        <input
          class="inputBase-0-2-2 inputBase-d1-0-2-4"
          readonly=""
          value="0"
        />
        <div
          class="wrapper-0-2-14 wrapper-d4-0-2-26"
          data-testid="ripple-wrapper"
          tabindex="-1"
        >
          <div
            class="rippleContainer-0-2-16 rippleContainer-d6-0-2-28"
          >
            <div
              class="ripple-0-2-15 ripple-d5-0-2-27 "
              data-testid="ripple-animation"
            />
          </div>
          <button
            class="button-0-2-7 button-d3-0-2-23"
            data-testid="onIncrement-btn"
            type="button"
          >
            <span
              class="labelContainer-0-2-8 labelContainer-d4-0-2-24"
            >
              <span
                class="label-0-2-9 label-d5-0-2-25"
              >
                +
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
